server/service/src/test_helpers.rs

```diff
@@ -131,30 +132,43 @@ pub(crate) fn make_movements(stock_line: StockLineRow, date_quantity: Vec<(i64,
             let invoice_id = format!("invoice_{}_{}_{}", stock_line.id, date, quantity);
             let date = Utc::now().naive_utc() + chrono::Duration::days(date - 30);
 
-            (
+            // Assuming external parties, non-transfers. So some dates are None
+            let invoice = if quantity > 0 {
                 InvoiceRow {
                     id: invoice_id.clone(),
                     store_id: stock_line.store_id.clone(),
                     name_link_id: mock_name_a().id.clone(),
-                    r#type: if quantity > 0 {
-                        InvoiceType::InboundShipment
-                    } else {
-                        InvoiceType::OutboundShipment
-                    },
-                    status: if quantity > 0 {
-                        InvoiceStatus::Verified
-                    } else {
-                        InvoiceStatus::Shipped
-                    },
-                    created_datetime: date,
-                    allocated_datetime: Some(date),
-                    picked_datetime: Some(date),
-                    shipped_datetime: Some(date),
-                    delivered_datetime: Some(date),
-                    received_datetime: Some(date),
-                    verified_datetime: Some(date),
+                    r#type: InvoiceType::InboundShipment,
+                    status: InvoiceStatus::Verified,
+                    created_datetime: date.with_hour(1).unwrap(),
```

Comment:
These dates generate the corresponding logs. So it is important that they're in the right order so that the latest activity log is actually the latest status in the sequence